command/harness/setup.go

```diff
@@ -214,6 +215,12 @@ func HandleSetup(ctx context.Context, r *SetupVMRequest, s store.StageOwnerStore
 		r.SetupRequest.MountDockerSocket = &b
 	}
 
+	deleteInfoMap := make(map[string]string)
+	deleteInfoMap["shouldDelete"] = "false"
+	if env.Runner.CleanupDebug {
+		util.CreateEnvState().Add(stageRuntimeID, deleteInfoMap)
+	}
+
 	setupResponse, err := client.Setup(ctx, &r.SetupRequest)
 	if err != nil {
 		go cleanUpFn(true)
```

Comment:
This will destroy the vm.

Reply:
this will ultimately go to manager.destroy. I have put the logic there to not destroy